authoringAssistant/src/main/java/explorableviz/transparenttext/LearningQueryContext.java

```diff
@@ -26,13 +28,18 @@ public static LearningQueryContext importLearningCaseFromJSON(String jsonLearnin
         String systemPrompt = jsonLearningCase.getString("system_prompt");
         ArrayList<QueryContext> learningCases = new ArrayList<>();
         Random random = new Random(0);
-
-        DirectoryStream<Path> directoryStream = Files.newDirectoryStream(Path.of(LEARNING_CASE_PATH), Files::isRegularFile);
-        for (Path filePath : directoryStream) {
-            TestQueryContext testQueryContext = TestQueryContext.importFromJson(filePath, random);
-            learningCases.addAll(testQueryContext.instantiate(numCasesToGenerate));
-        }
-
+        Files.list(Paths.get(LEARNING_CASE_PATH))
+                .filter(Files::isRegularFile)
```

Comment:
The indentation is a bit odd here – maybe thinking about using one rather than levels of indentation for the `.filter` part and then two rather than one for the `try` block? (There is no right or wrong here obviously but the positioning of the `try` here is a bit tricky.)